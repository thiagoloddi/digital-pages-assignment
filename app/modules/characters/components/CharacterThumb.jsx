import React from 'react';
import injectSheet from 'react-jss';
import Image from '../../../components/ui/Image';


const style = {
  container: {
    height: '50px',
    '& > *': {
      display: 'inline-block',
      verticalAlign: 'middle'
    }
  },
  name: {
    maxWidth: 'calc(100% - 50px)',
    paddingLeft: '10px'
  }
};

const CharacterThumb = ({ name, classes, src }) => {
  return (
    <div className={classes.container}>
      <Image src={src}/>
      <div className={classes.name}>{name}</div>
    </div>
  )
}

export default injectSheet(style)(CharacterThumb);