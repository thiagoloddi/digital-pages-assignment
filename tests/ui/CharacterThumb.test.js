import React from 'react';
import renderer from 'react-test-renderer';
import CharacterThumb from '../../app/modules/characters/components/CharacterThumb';

const createNodeMock = e => {
  if(e.type == 'img') {
    return {};
  }
};

it('renders correctly', () => {
  const tree = renderer
    .create(<CharacterThumb name="Char" src="path.jpg" />, { createNodeMock })
    .toJSON();
  expect(tree).toMatchSnapshot();
});