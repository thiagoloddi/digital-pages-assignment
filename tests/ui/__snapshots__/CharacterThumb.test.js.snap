// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="CharacterThumb-container-0-1-1"
>
  <div
    className="Image-container-0-1-3 "
  >
    <img
      className="Image-img-0-1-4"
      src="path.jpg"
      style={
        Object {
          "visibility": "hidden ",
        }
      }
    />
    <img
      className="Image-loading-0-1-5"
      src="/assets/progress.gif"
      style={null}
    />
  </div>
  <div
    className="CharacterThumb-name-0-1-2"
  >
    Char
  </div>
</div>
`;
